pkg/authorization/cache/readonlycache.go

```diff
@@ -0,0 +1,136 @@
+	c.ReadOnlyClusterPolicyBindingCache.RunUntil(bindingStopChannel)
+}
+
+// NewReadOnlyCache creates a new ReadOnlyCache.  You cannot use a normal client, because you don't want policy guarding the policy from the authorizer
+func NewReadOnlyCache(bindingRegistry bindingregistry.WatchingRegistry, policyRegistry policyregistry.WatchingRegistry,
+	clusterBindingRegistry clusterbindingregistry.WatchingRegistry, clusterPolicyRegistry clusterpolicyregistry.WatchingRegistry) *ReadOnlyCache {
+	return &ReadOnlyCache{
+		ReadOnlyPolicyCache:               NewReadOnlyPolicyCache(policyRegistry),
+		ReadOnlyClusterPolicyCache:        NewReadOnlyClusterPolicyCache(clusterPolicyRegistry),
+		ReadOnlyPolicyBindingCache:        NewReadOnlyPolicyBindingCache(bindingRegistry),
+		ReadOnlyClusterPolicyBindingCache: NewReadOnlyClusterPolicyBindingCache(clusterBindingRegistry),
+	}
+}
+
+// LastSyncResourceVersion exposes the LastSyncResourceVersion of the internal reflectors from each cache - conforms to the standard that
+// a SyncResourceVersion is a monotonically increasing integer
+func (c ReadOnlyCache) LastSyncResourceVersion() string {
+	resourceVersions := []string{}
+	resourceVersions = append(resourceVersions, c.ReadOnlyPolicyCache.LastSyncResourceVersion())
+	resourceVersions = append(resourceVersions, c.ReadOnlyClusterPolicyCache.LastSyncResourceVersion())
+	resourceVersions = append(resourceVersions, c.ReadOnlyPolicyBindingCache.LastSyncResourceVersion())
+	resourceVersions = append(resourceVersions, c.ReadOnlyClusterPolicyBindingCache.LastSyncResourceVersion())
+
+	currentState := strings.Join(resourceVersions, "")
+
+	return currentState
+}
+
+// Methods below conform the ReadOnlyCache to the ReadOnlyPolicyClient interface
+
+func (c ReadOnlyCache) ReadOnlyPolicies(namespace string) client.ReadOnlyPolicyInterface {
+	return newReadOnlyPolicies(c, namespace)
+}
+
+func (c ReadOnlyCache) ReadOnlyPolicyBindings(namespace string) client.ReadOnlyPolicyBindingInterface {
+	return newReadOnlyPolicyBindings(c, namespace)
+}
+
+func (c ReadOnlyCache) ReadOnlyClusterPolicies() client.ReadOnlyClusterPolicyInterface {
+	return newReadOnlyClusterPolicies(c)
+}
+
+func (c ReadOnlyCache) ReadOnlyClusterPolicyBindings() client.ReadOnlyClusterPolicyBindingInterface {
+	return newReadOnlyClusterPolicyBindings(c)
+}
+
+// Calls to GetPolicy or ListPolicyBindings pass an empty string as the namespace to refer to cluster policies or bindings.
+// This conflicts with kapi.NamespaceAll, and calls that do not route through GetPolicy() (but rather ReadOnlyPolicies().Get()),
+// or through ReadOnlyPolicyBindings().List() instead of ListPolicyBindings() assume that namespace == "" is equivalent to
+// namespace == kapi.NamespaceAll. Calls that route through the following methods assume namespace == "" refers to cluster to
+// support the rulevalidation.PolicyGetter and rulevalidation.BindingLister interfaces.
+func namespaceRefersToCluster(namespace string) bool {
```

Comment:
It is not necessary for unexported functions though it's a good practice to do it